store/depositDates.js

```diff
@@ -11,21 +12,34 @@ class DepositDates {
     this.rootStore = rootStore
   }
 
-  retrieveDepositDates = async (pageNumber, searchTerm) => {
-    const key = `${this.rootStore.dataProvider}-${pageNumber}-${searchTerm}`
+  retrieveDepositDates = async (pageNumber, searchTerm, columnOrder) => {
+    const order = getOrder(columnOrder)
+    const key = `${this.rootStore.dataProvider}-${pageNumber}-${searchTerm}-${order}`
     // TODO: Invalidate cache after some time
     //       Move to @oacore/api
     if (this.pages.has(key)) return this.pages.get(key)
-    const [data] = await apiRequest(
-      `/data-providers/${this.rootStore.dataProvider}/public-release-dates`,
-      'GET',
-      {
-        from: pageNumber * PAGE_SIZE,
-        size: PAGE_SIZE,
-        q: searchTerm,
-      },
-      true
-    )
+
+    const params = {
+      from: pageNumber * PAGE_SIZE,
+      size: PAGE_SIZE,
+    }
+
+    if (order) params.orderBy = order
+    if (searchTerm) params.q = searchTerm
+
+    let data
+    try {
+      ;[data] = await apiRequest(
```

Comment:
Why API request does return an array?

Reply:
Because I designed it that way. 😄 It was not a clever move. Anyway, I prepared a refactor of it...coming in following PR.